custom_model_runner/datarobot_drum/drum/language_predictors/r_predictor/r_predictor.py

```diff
@@ -104,3 +104,68 @@ def predict(self, input_filename):
                 logger.error(error_message)
                 raise DrumCommonException(error_message)
         return py_data_object
+
+    def predict_unstructured(self, data, **kwargs):
+        def _r_is_character(r_val):
+            _is_character = ro.r("is.character")
+            return bool(_is_character(r_val))
+
+        def _r_is_raw(r_val):
+            _is_raw = ro.r("is.raw")
+            return bool(_is_raw(r_val))
+
+        def _r_is_null(r_val):
+            _is_null = ro.r("is.null")
+            return bool(_is_null(r_val))
+
+        def _cast_r_to_py(r_val):
+            if _r_is_null(r_val):
+                return None
+            elif _r_is_raw(r_val):
+                return bytes(r_val)
+            elif _r_is_character(r_val):
+                # Any scalar value is returned from R as one element vector,
+                # so get this value.
+                return str(r_val[0])
+            else:
+                raise DrumCommonException(
+                    "Can not convert R value {} type {}".format(r_val, type(r_val))
+                )
+
+        def _rlist_to_dict(rlist):
+            if _r_is_null(rlist):
+                return None
+            return {str(k): _cast_r_to_py(v) for k, v in rlist.items()}
+
+        data_binary_or_text = data
+
+        if UnstructuredDtoKeys.QUERY in kwargs:
+            if isinstance(kwargs[UnstructuredDtoKeys.QUERY], dict):
+                query_kwargs = kwargs[UnstructuredDtoKeys.QUERY]
+                for k, v in query_kwargs.items():
+                    if v is None:
+                        query_kwargs[k] = ro.rinterface.NULL
```

Comment:
there's no point passing Null values to R since they are unavailable as keys at all there, they should be skipped afaik

Reply:
also query arg value likely cant be None

Reply:
You are right.
Not relevant, after I changed parser.
This code is removed.